kotlin/offline-routing/src/main/java/com/esri/arcgisruntime/sample/offlinerouting/MainActivity.kt

```diff
@@ -0,0 +1,270 @@
+    val basemap = Basemap(tiledLayer)
+    mapView.map = ArcGISMap(basemap)
+
+    // add the graphics overlays to the map view
+    mapView.graphicsOverlays.addAll(listOf(stopsOverlay, routeOverlay))
+
+    // load the route task
+    routeTask.loadAsync()
+    routeTask.addDoneLoadingListener {
+      if (routeTask.loadStatus == LoadStatus.LOADED) {
+        try {
+          // create route parameters
+          routeParameters = routeTask.createDefaultParametersAsync().get()
+        } catch (e: Exception) {
+          when (e) {
+            is InterruptedException, is ExecutionException -> Log.e(
+              TAG,
+              "Error getting default route parameters. ${e.stackTrace}"
+            )
+          }
+        }
+      } else {
+        Log.e(TAG, "Error loading route task. ${routeTask.loadError.message}")
+      }
+    }
+
+    // set up travel mode switch
+    modeSwitch.setOnCheckedChangeListener { _, isChecked ->
+      routeParameters?.travelMode = when (isChecked) {
+        true -> routeTask.routeTaskInfo.travelModes[0]
+        false -> routeTask.routeTaskInfo.travelModes[1]
+      }
+      Toast.makeText(
+        this,
+        "${routeParameters?.travelMode?.name} route selected.",
+        Toast.LENGTH_SHORT
+      ).show()
+      updateRoute()
+    }
+
+    // make a clear button to reset the stops and routes
+    clearButton.setOnClickListener {
+      stopsOverlay.graphics.clear()
+      routeOverlay.graphics.clear()
+    }
+
+    // add a graphics overlay to show the boundary
+    GraphicsOverlay().let {
+      val envelope = Envelope(
+        Point(-13045352.223196, 3864910.900750, 0.0, SpatialReferences.getWebMercator()),
+        Point(-13024588.857198, 3838880.505604, 0.0, SpatialReferences.getWebMercator())
+      )
+      val boundarySymbol = SimpleLineSymbol(SimpleLineSymbol.Style.DASH, 0xFF00FF00.toInt(), 5f)
+      it.graphics.add(Graphic(envelope, boundarySymbol))
+      mapView.graphicsOverlays.add(it)
+    }
+
+    // set up the touch listeners on the map view
+    createMapGestures()
+  }
+
+  /**
+   * Sets up the onTouchListener for the mapView.
+   * For single taps, graphics will be selected.
+   * For double touch drags, graphics will be moved.
+   * */
+  private fun createMapGestures() {
+    mapView.onTouchListener = object : DefaultMapViewOnTouchListener(this, mapView) {
+      override fun onSingleTapConfirmed(motionEvent: MotionEvent): Boolean {
+
+        val screenPoint = android.graphics.Point(
+          motionEvent.x.roundToInt(),
+          motionEvent.y.roundToInt()
+        )
+        addOrSelectGraphic(screenPoint)
+        return true
+      }
+
+      override fun onDoubleTouchDrag(motionEvent: MotionEvent): Boolean {
+
+        val screenPoint = android.graphics.Point(
+          motionEvent.x.roundToInt(),
+          motionEvent.y.roundToInt()
+        )
+        // UNSURE: should you be able to select a graphic with double touch drag?
```

Comment:
the graphic needs to be selected by a single tap before it can be moved with a double tap. it stays selected after the move and so this isn't necessary.

Reply:
okay, I'll leave it out if you found it easy enough to control, then. Thanks!